packages/clerk-js/src/ui/components/UserProfile/DeletePage.tsx

```diff
@@ -0,0 +1,44 @@
+import { useCoreClerk, useCoreSession, useEnvironment, useCoreUser } from '../../contexts';
+import { localizationKeys, Text } from '../../customizables';
+import { ContentPage, Form, FormButtons, SuccessPage, useCardState, withCardStateProvider } from '../../elements';
+import { handleError } from '../../utils';
+import { UserProfileBreadcrumbs } from './UserProfileNavbar';
+import { useRouter } from '../../router';
+
+export const DeletePage = withCardStateProvider(() => {
+  const card = useCardState();
+  const environment = useEnvironment();
+  const router = useRouter();
+  const clerk = useCoreClerk();
+  const session = useCoreSession();
+  const user = useCoreUser();
+
+  const deleteUser = async () => {
+    try {
+      await user.delete();
+      await clerk.signOut({ sessionId: session.id });
+      router.navigate(environment.displayConfig.homeUrl);
+      // TODO: for routerless mode, explicitly open the "choose" route
+      clerk.openSignIn();
```

Comment:
Want to add a little extra context here - after deleting a user, you get signed out, and the router will navigate to the home route, which should contain the sign in UI, in theory. If you have a multi-session app, the sign in UI should allow the user to select the next available session data to log into.

However, if the implementation is just using pure JS and is not associated with a router, the navigation will do nothing other than changing the url hash, so we explicitly open the sign in window. This is a slight bug IMO for router-less apps, since we want to show the "choose" route within the sign in UI to allow to user to choose other sessions if there are any. If you refresh the page, it will automatically log into any other available sessions for a multi-session app, but a refresh shouldn't be necessary.

Reply:
Is this fix targeted for multi-session or single-session apps? The sign-in choose screen is only available in multi-session apps.

This behavior is opinionated when it comes to UX. ClerkJS supports [afterSignOut props](https://github.com/clerkinc/javascript/blob/24915e26581a39651f389ded9d1ae0c6533e167f/packages/clerk-js/src/ui/components/UserButton/useMultisessionActions.tsx#L27) that should also be used in this case as they are the main control for Clerk users to handle post sign-out behavior.

Most importantly, I'd also that FAPI signs out the user during `await user.delete`. So, the `clerk.signOut` shouldn't be necessary as the payload of `user.delete` should purge the current session, and <UserProfile/> should unmount. Let me double-check that.